### PR TITLE
[WIP] fix tests

### PR DESCRIPTION
**Pull request by sonnyp**

### CODE REVIEW

File: test/issues.js [hunk 1]
```diff
@@ -65,26 +67,22 @@ describe('Stream without proper "to" attribute', function() {
...
                 }
             }
-            done(error)
         })
 
         it('Should close stream', function(done) {
             var end = new Date().getTime() + 2000
-            var error = null
-            while (streamClosed !== true) {
+            while (streamClosed === true) {
```
Comment (sonnyp) [inline test/issues.js]: not sure about this, @chris-rock ?

Comment (chris-rock) [inline test/issues.js]: hey @sonnyp calling done in a loop may lead to multiple callbacks.

Comment (Retsam) [inline test/issues.js]: The new logic both here, and in the previous `it` block looks incorrect to me.

Old version:
```
while(!(conditionWeExpectToEventuallyBeTrue)) {
	if(timeoutHasExpired) {
		//indicate that we've timed out
		break;
	}
}
//callback, possibly with error
```
New version:
```
while(conditionWeExpectToEventuallyBeTrue) {
	if(timeoutHasExpired) {
		passTest()
	}
}
//do nothing, probably gets caught by the test runner's default timeout logic
```

The key is that `streamClosed` is something that the test expects might not initially be true, but should eventually become true, so looping until it isn't true doesn't make sense.

**Comment by chris-rock** [comment]
@sonnyp how far are we here? I would love to merge some PRs.

**Comment by sonnyp** [comment]
@chris-rock no idea, will get this done this week.

**Comment by chris-rock** [comment]
cool

**Pull request closed without merging**